parse_events/handle_contracts.py

```diff
@@ -54,7 +54,7 @@ def _read_ipfs_node(self, ipfs_hash):
         return json.loads(ipfs_data.decode('utf8'))
 
     def _hex_to_str(self, hex_str):
-        return Web3.toText(hex_str.rstrip("0"))
+        return Web3.toText(hex_str).rstrip("\u0000")
```

Comment:
please do check if there are utilities to remove padding from data in bytes retrieved from block chain.
approving this for now based on our discussion.